### ISSUE TITLE
Capture only the last X Seconds before key is pressed

### ISSUE THREAD

**Issue by ovidiubarbu** [MathewSachin/Captura#182]
Hi Mathew,

I really like the tool, and comes quite in handy! I would like to propose a new feature, which I think it would be helpful to have.

Have the following options:
- Always capture (this means as soon as the tool starts, it will start capturing based on your default settings)
    - Capture only the last X Seconds before the Start/Stop capture key is pressed  (this is a sub-option of the above)

Basically this will capture the last, let's say, 30 seconds before I pressed ALT-F9 key.

Thank you!

**Comment by MathewSachin** [comment]
I looked for information on how this could be done but found nothing. Seems like this will not be possible currently.
The main problem is how to maintain the file like a circular buffer.

One way that I thought is to save frames as images in a folder, deleting images older than X seconds, and encode the images into video when required. But, I doubt about the efficiency of this process.

**Comment by ovidiubarbu** [comment]
This is the only way I see it too for now. Save chunks of 1 sec maybe, and stick them into final video file. I saw that, right now, we add frame to frame to the video stream, but could not find a method to remove a frame or more from the beginning of the stream. That would be helpful!
